cuqi/experimental/_recommender.py

```diff
@@ -0,0 +1,200 @@
+import cuqi
+import inspect
+import numpy as np
+
+# This import makes suggest_sampler easier to read
+import cuqi.experimental.mcmc as samplers 
+
+
+class SamplerRecommender(object):
+    """
+    This class can be used to automatically choose a sampler.
+
+    Parameters
+    ----------
+    target: Density or JointDistribution
+        Distribution to get sampler recommendations for.
+
+    exceptions: list[cuqi.experimental.mcmcm.Sampler], *optional*
+        Samplers not to be recommended.
+    """
+
+    def __init__(self, target:cuqi.density.Density, exceptions = []):
+        self._target = target
+        self._exceptions = exceptions
+        self._create_ordering()
+
+    @property
+    def target(self) -> cuqi.density.Density:
+        """ Return the target density. """
```

Comment:
Distribution instead of Density?
```suggestion
        """ Return the target Distribution. """
```